pkg/flare/archive_nix.go

```diff
@@ -7,10 +7,83 @@
 
 package flare
 
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
 func zipCounterStrings(tempDir, hostname string) error {
 	return nil
 }
 
 func zipTypeperfData(tempDir, hostname string) error {
 	return nil
 }
+
+// initPermsInfo creates the permissions.log file and write the headers.
+func initPermsInfo(tempDir, hostname string, p os.FileMode) error {
```

Comment:
As I mentioned above, I would probably use a struct, and use this to initialize a map for our file permissions, and destination file, then use the `addPermsInfo` to populate the map. And finally some `commitPermsInfo()` function or similar to write all that info to the file. What do you think?

Reply:
Yes the `commit` solution looks like the good one. Although, I'm wondering if we should invalidate the "permissions struct" if something wrong happened during the stat parts, or going with a best effort and write as much permissions info as we can (what we were doing in the current implementation).